langchain/src/chat_models/iflytek_xinghuo.ts

```diff
@@ -0,0 +1,475 @@
+      );
+      const authorization_origin = `api_key="${this.iflytekApiKey}", algorithm="hmac-sha256", headers="host date request-line", signature="${hash}"`;
+      authorization = window.btoa(authorization_origin);
+    } else {
+      throw new Error(`Invalid Environment: ${getEnv()}`);
+    }
+    let authWebSocketUrl = this.apiUrl;
+    authWebSocketUrl += `?authorization=${authorization}`;
+    authWebSocketUrl += `&host=${encodeURIComponent(host)}`;
+    authWebSocketUrl += `&date=${encodeURIComponent(date)}`;
+    return authWebSocketUrl;
+  }
+
+  /**
+   * Get the identifying parameters for the model
+   */
+  identifyingParams() {
+    return {
+      version: this.version,
+      ...this.invocationParams(),
+    };
+  }
+
+  /**
+   * Get the parameters used to invoke the model
+   */
+  invocationParams(): Omit<ChatCompletionRequest, "messages"> {
+    return {
+      temperature: this.temperature,
+      top_k: this.top_k,
+    };
+  }
+
+  /** @ignore */
+  async completion(
+    request: ChatCompletionRequest,
+    onmessage: (data: string) => void,
+    signal?: AbortSignal
+  ) {
+    const url = await this.getAuthWebSocketUrl();
+    const webSocketStream = new WebSocketStream(url, {
+      signal,
+    });
+    const connection = await webSocketStream.connection;
+    const header = {
+      app_id: this.iflytekAppid,
+      uid: this.userId,
+    };
+    const parameter = {
+      chat: {
+        domain: this.domain,
+        temperature: request.temperature ?? this.temperature,
+        max_tokens: request.max_tokens ?? this.max_tokens,
+        top_k: request.top_k ?? this.top_k,
+      },
+    };
+    const payload = {
+      message: {
+        text: request.messages,
+      },
+    };
+    const message = JSON.stringify({
+      header,
+      parameter,
+      payload,
+    });
+    const { writable, readable } = connection;
+    const writer = writable.getWriter();
+    await writer.write(message);
+    const streams = IterableReadableStream.fromReadableStream(readable);
+    for await (const chunk of streams) {
+      onmessage?.(chunk as string);
+      const { header } = JSON.parse(chunk as string);
+      if (header.status === 2) {
+        break;
+      }
+    }
+    void streams.cancel();
+    void webSocketStream.close();
+  }
+
+  /** @ignore */
+  async _generate(
+    messages: BaseMessage[],
+    options: this["ParsedCallOptions"],
+    runManager?: CallbackManagerForLLMRun | undefined
+  ): Promise<ChatResult> {
+    const tokenUsage: TokenUsage = {};
+    const params = this.invocationParams();
+    const messagesMapped: XinghuoMessage[] = messages.map((message) => ({
+      role: messageToXinghuoRole(message),
+      content: message.content,
+    }));
+    const data = await new Promise<ChatCompletionResponse>(
+      (resolve, reject) => {
+        let response: ChatCompletionResponse;
+        let rejected = false;
+        let resolved = false;
+        void this.completion(
```

Comment:
This `void` needs a linter ignore line

Reply:
And actually, it would be better to refactor `this.completion` to either not return a promise or be an iterator

Reply:
I hava refactored completion method to either not return a promise or be an iterator.